### PR TITLE
Azure blob storage artifact support

### PR DESCRIPTION
**Pull request by akshayaccha**
https://github.com/argoproj/argo/pull/874 Changes from this pr have been taken and changes with respect to comments have been made

### CODE REVIEW

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -705,6 +709,43 @@
         }
       }
     },
+    "io.argoproj.workflow.v1alpha1.AzureBlobArtifact": {
+      "description": "AzureBlobArtifact is the location of an Azurer Blob Storage artifact",
```
Comment (AnesBenmerzoug) [inline api/openapi-spec/swagger.json]: I think ```Azurer``` in ```"AzureBlobArtifact is the location of an Azurer Blob Storage artifact"``` is a typo and should be ```Azure``` instead

**Comment by AnesBenmerzoug** [comment]
It seems the checks failed because of linting issues.
You should running ```gofmt``` and ```goimports``` on the changed files

**Comment by logicfox** [comment]
Is this being worked on? We'd love to have this feature shipped.

File: workflow/artifacts/azureblob/azureblob.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
+		return nil, err
+	}
+
+	p := azblob.NewPipeline(credentials, azblob.PipelineOptions{})
+
+	u, err := url.Parse(fmt.Sprintf("https://%s.%s/%s", azblobDriver.AccountName, azblobDriver.Key, azblobDriver.Container))
+
+	log.Info(azblobDriver.AccountName)
```
Comment (alexec) [inline workflow/artifacts/azureblob/azureblob.go]: should be debug level?

File: workflow/artifacts/azureblob/azureblob.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	Key                      string
+	AccountName              string
+	AccountKey               string
+}
+
+// newMinioClient instantiates a new minio client object.
+func (azblobDriver *AzureBlobArtifactDriver) newAzureBlobClient() (*azblob.ContainerURL, error) {
+	credentials, err := azblob.NewSharedKeyCredential(azblobDriver.AccountName, azblobDriver.AccountKey)
+	if err != nil {
+		log.Fatal("Invalid credentials with error: " + err.Error())
```
Comment (alexec) [inline workflow/artifacts/azureblob/azureblob.go]: should not log here?

File: workflow/artifacts/azureblob/azureblob.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+// Load downloads artifacts from Azure Blob Storage
+func (azblobDriver *AzureBlobArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
+
+	containerUrl, err := azblobDriver.newAzureBlobClient()
+	if err != nil {
+		return errors.InternalWrapError(err)
+	}
+	ctx := context.Background()
+	log.Info(azblobDriver.EndpointSuffix)
```
Comment (alexec) [inline workflow/artifacts/azureblob/azureblob.go]: should be debug level?

File: workflow/artifacts/azureblob/azureblob.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	}
+	defer file.Close()
+
+	r, err := azblob.UploadFileToBlockBlob(ctx, file, blobURL, azblob.UploadToBlockBlobOptions{
+		BlockSize:   4 * 1024 * 1024,
+		Parallelism: 16})
+	if err != nil {
+		return errors.InternalWrapError(err)
+	}
+	log.Info(r)
```
Comment (alexec) [inline workflow/artifacts/azureblob/azureblob.go]: should be debug level?

**Comment by logicfox** [comment]
Bump! :)

**Comment by simster7** [comment]
@akshayaccha Do you plan on finishing up this PR? If not, the community might be able to help out

**Comment by sinvalmendes** [comment]
Would be very nice to get this done

**Comment by simster7** [comment]
@sinvalmendes Or anybody else interested feel free to take over this PR. I would recommend starting from scratch though, instead of dealing with the merge conflicts.

**Comment by alexec** [comment]
I think we should close this PR.

**Comment by logicfox** [comment]
I try to take a stab at this issue this weekend.

**Comment by simster7** [comment]
Closing in favor of #3173

**Comment by jaimeamesti** [comment]
Hi,

I'm a newbie on open source projects. I really love this feature and I would like to finish this PR. Can anyone tell me how to reopen this PR?

**Comment by agilgur5** [comment]
This was superseded by #2515 , #3173 , and eventually https://github.com/argoproj/argo-workflows/pull/9026

**Pull request closed without merging**